src/plugins/data/common/data_frames/utils.ts

```diff
@@ -472,3 +472,77 @@ export const dataFrameToSpec = (dataFrame: IDataFrame, id?: string): IndexPatter
     fields: fields.reduce(flattenFields, {} as IndexPatternFieldMap),
   };
 };
+
+type FetchFunction<T, P = void> = (params?: P) => Promise<T>;
+
+export interface PollingConfigurations {
+  tabId: string;
+}
+
+export class UsePolling<T, P = void> {
```

Comment:
i think this could extend the partial data frame if not then this could fits more of a function. 

or avoid the usage of `Use` to avoid confusion with https://react.dev/reference/react/use